### PR TITLE
Feature/es api 5.0 v2

### PR DESCRIPTION
**Pull request by wolfgang-s**
This is now for the develop branch.

Some small changes to make it compatible again and  remove conflicts.

### CODE REVIEW

**Comment by jamesjara** [comment]
we need this..

**Comment by wolfgang-s** [comment]
@aquid please merge this. small changes -> huge impact :) ... all tests should pass!

When this is merged I can continue submitting PRs with additional features.

Thanks!

**Comment by saurabh73** [comment]
Need this connector to support v5.

**Comment by saurabh73** [comment]
Checks failing due to mismatch in $PACKAGE_NAME variable in [CI Config](http://ci.strongloop.com/job/loopback-connector-elastic-search/node=6.x,os=windows/806/parameters/) and **package.json**

[sl-ci-run ERROR 49.98 ]: Package name: loopback-connector-es does not match $PACKAGE_NAME: loopback-connector-elastic-search

**Comment by saurabh73** [comment]
`eslint **/*.js

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\common\models\user-model.js
   7:47  error  'obj' is defined but never used     no-unused-vars
  13:38  error  'target' is defined but never used  no-unused-vars

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\boot\explorer.js
   6:9  error  Unexpected console statement  no-console
  21:9  error  Unexpected console statement  no-console

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\roles\01-create-role.js
   8:5  error  '_' is defined but never used        no-unused-vars
   9:5  error  'Promise' is defined but never used  no-unused-vars
  39:9  error  'users' is defined but never used    no-unused-vars

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\roles\02-findOrCreate-role.js
   8:5  error  '_' is defined but never used        no-unused-vars
   9:5  error  'Promise' is defined but never used  no-unused-vars
  39:9  error  'users' is defined but never used    no-unused-vars

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\server.js
  36:9  error  Unexpected console statement  no-console

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\users\01-find-user.js
   8:5   error  '_' is defined but never used  no-unused-vars
  52:21  error  Unexpected console statement   no-console
  62:13  error  Unexpected console statement   no-console

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\users\02-findOrCreate-user.js
   8:5   error  '_' is defined but never used  no-unused-vars
  55:21  error  Unexpected console statement   no-console
  65:13  error  Unexpected console statement   no-console

D:\workspace\loopback-connector-elastic-search\d60c4a15\examples\server\users\03-create-user.js
   8:5   error  '_' is defined but never used  no-unused-vars
  50:21  error  Unexpected console statement   no-console
  60:13  error  Unexpected console statement   no-console

D:\workspace\loopback-connector-elastic-search\d60c4a15\lib\esConnector.js
  784:2  error  Mixed spaces and tabs  no-mixed-spaces-and-tabs

? 21 problems (21 errors, 0 warnings)`

Fix the specified errors for Pretest failures in 
**Failed at the loopback-connector-es@1.3.5 pretest script 'eslint **/*.js'.**

**Comment by wolfgang-s** [comment]
@saurabh73 These checks are currently ignored and need to be fixed in a separate PR.

This PR is from my point of view ready to be merged ... @aquid

**Comment by saurabh73** [comment]
@wolfgang-s I hope this get's merged too soon and get's in next release... Till then, will work with your repo

**Comment by aquid** [comment]
@pulkitsinghal Please have a look in this PR and provide me with your feedback. I will take your feedback and then will either merge it or ask for necessary changes. 

@wolfgang-s I would need some feedback from @pulkitsinghal before merging this PR. You see, `suggest search` was added for some external request and I need to be sure that this PR doesn't break anything existing.

**Comment by pulkitsinghal** [comment]
Hello @wolfgang-s and @aquid, here's my opinion:
1. I will have to modify this PR.
2. I will not remove `suggest` because it has an `if` block around it so it cannot cause issues in 5.x unless someone shoots themselves in the foot by adding it to the criteria.
3. With the changes i make, 1.x and 2.x users will continue to be able to use `suggest` but there will be no new work to help 5.x users utilize it because we haven't done the work to pick it off the criteria body in 5.x! A separate PR should be created to bring 5.x up to snuff for those who want to use `suggest`.

Ok starting work now.

**Comment by wolfgang-s** [comment]
@pulkitsinghal just to make sure, the suggest tests still work, because I use suggest using the search endpoint. Why use the deprecated suggest endpoint? (ok - it was deprecated in v5.0)

So, I do not see any reason to add this call to the suggest endpoint?

Maybe I got something wrong, but obviously all tests pass? even the suggest tests? ... ok - maybe the tests are not good enough, I don't know. Maybe we are missing specific tests?

I think we should stick to the approach having one test set for all ES versions, if you add now again the suggest endpoint, you will see for ES 5 the tests won't pass (ES syntax is different).

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     "async": "^0.9.0",
     "bluebird": "^2.9.14",
     "debug": "^2.1.3",
-    "elastic-deletebyquery": "1.0.7",
-    "elasticsearch": "^11.0.1",
+    "elastic-deletebyquery": "^1.0.8",
```
Comment (pulkitsinghal) [inline package.json]: @wolfgang-s - is an update from 1.0.7 to 1.0.8 required for working with 5.x? or if its for another reason, can it be part of a separate PR?

Comment (wolfgang-s) [inline package.json]: No need to upgrade "elastic-deletebyquery", this is true ... I just check the versions and I thought why not used the latest? :) -> you can remove it

File: test/02.basic-querying.test.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ var db, User, Customer, AccessToken, Post, PostWithId, Category, SubCategory;
 
 /*eslint no-console: "off"*/
 /*global getSchema should*/
-describe('basic-querying', function () {
+xdescribe('basic-querying', function () {
```
Comment (pulkitsinghal) [inline test/02.basic-querying.test.js]: Why was this turned off and committed?
`+xdescribe('basic-querying', function () {`
cc @aquid @wolfgang-s

**Comment by pulkitsinghal** [comment]
Hello @wolfgang-s 
1. After working on this a bit, my understanding is better. Not everything I said in my most recent comment was valid.
2. I pulled your repo into a [branch](https://github.com/strongloop-community/loopback-connector-elastic-search/tree/wolfgang-s-feature/es-api-5.0-v2) so i may work on it:
  ```
  git checkout -b wolfgang-s-feature/es-api-5.0-v2 develop
  git pull git://github.com/wolfgang-s/loopback-connector-elastic-search.git feature/es-api-5.0-v2
  ```
3. So please have a look at the most relevant [commit](https://github.com/strongloop-community/loopback-connector-elastic-search/commit/81cac552c191505d7e6e34c09dea35e7a92a73b8) in there that was made on top of your stuff. If you and @aquid agree that my changes make sense then by all means go ahead and finish this thing off and release it, I won't stand in your way :)

File: test/02.basic-querying.test.js (hunk 2 above)
Comment (wolfgang-s) [inline test/02.basic-querying.test.js]: @pulkitsinghal since I should not change existing tests, I copied the basic-querying tests and removed all timeouts (since my last PR regarding refresh: true this is possible)
- 03.basic-querying-no-timeout.test.js
The tests are the same as the other tests but are way faster ... if you like, you can reenable them or remove them completely (my suggestion)

Comment (aquid) [inline test/02.basic-querying.test.js]: @wolfgang-s I won't turn it off, it make sense to leave it active as it doesn't cause any problem because with both refresh as true and false these test cases would pass successfully. So before merging this I would revert it and make it as active.

**Comment by aquid** [comment]
@pulkitsinghal I can't see your updates in this PR, so I have created another [PR](https://github.com/strongloop-community/loopback-connector-elastic-search/pull/89) including your changes. Let me know if those changes are correct.

cc @wolfgang-s
